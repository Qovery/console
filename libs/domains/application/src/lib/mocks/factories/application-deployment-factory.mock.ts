import { DeploymentHistoryApplication, StateEnum } from 'qovery-typescript-axios'
import { Chance } from 'chance'

const chance = new Chance()

export const applicationDeploymentsFactoryMock = (howMany: number): DeploymentHistoryApplication[] =>
  Array.from({ length: howMany }).map((_, index) => ({
    id: index.toString(),
    created_at: new Date().toString(),
    updated_at: new Date().toString(),
    name: chance.name(),
    status: chance.pickone(Object.values([StateEnum.DEPLOYED, StateEnum.RUNNING, StateEnum.STOP_ERROR])),
  }))
