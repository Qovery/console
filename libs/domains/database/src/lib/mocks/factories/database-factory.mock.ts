import { Chance } from 'chance'
import { Database, DatabaseTypeEnum, DatabaseModeEnum, DatabaseAccessibilityEnum } from 'qovery-typescript-axios'

const chance = new Chance()

export const databaseFactoryMock = (howMany: number): Database[] =>
  Array.from({ length: howMany }).map((_, index) => ({
    id: `${index}`,
    created_at: new Date().toString(),
    updated_at: new Date().toString(),
    name: chance.name(),
    type: DatabaseTypeEnum.POSTGRESQL,
    version: '12',
    mode: DatabaseModeEnum.CONTAINER,
    accessibility: DatabaseAccessibilityEnum.PRIVATE,
    cpu: 1000,
    memory: 1024,
    storage: 1024,
    environment: {
      id: chance.guid(),
    },
    host: chance.name(),
    port: 80,
    maximum_cpu: 10,
    maximum_memory: 10,
    disk_encrypted: false,
  }))
