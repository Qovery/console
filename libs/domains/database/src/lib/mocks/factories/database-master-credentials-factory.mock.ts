import { Chance } from 'chance'
import { Credentials } from 'qovery-typescript-axios'

const chance = new Chance()

export const databaseMasterCredentialsFactoryMock = (howMany: number): Credentials[] =>
  Array.from({ length: howMany }).map((_, index) => ({
    login: chance.name(),
    password: chance.guid(),
    host: chance.name(),
    port: chance.d4(),
  }))
