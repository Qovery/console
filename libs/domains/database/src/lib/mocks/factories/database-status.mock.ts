import { Chance } from 'chance'
import { Status, ServiceDeploymentStatusEnum, StateEnum } from 'qovery-typescript-axios'

const chance = new Chance()

export const databaseStatusFactoryMock = (howMany: number): Status[] =>
  Array.from({ length: howMany }).map((_, index) => ({
    id: `${index}`,
    message: chance.sentence(),
    service_deployment_status: chance.pickone(
      Object.values([
        ServiceDeploymentStatusEnum.UP_TO_DATE,
        ServiceDeploymentStatusEnum.UP_TO_DATE,
        ServiceDeploymentStatusEnum.UP_TO_DATE,
      ])
    ),
    state: chance.pickone(Object.values([StateEnum.DEPLOYED, StateEnum.RUNNING, StateEnum.STOP_ERROR])),
  }))
