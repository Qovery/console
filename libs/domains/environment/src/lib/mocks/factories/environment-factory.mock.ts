import { Chance } from 'chance'
import {
  DeploymentHistoryStatusEnum,
  EnvironmentModeEnum,
  ServiceDeploymentStatusEnum,
  StateEnum,
  WeekdayEnum,
} from 'qovery-typescript-axios'
import { EnvironmentEntity } from '@qovery/shared/interfaces'

const chance = new Chance()

const service = {
  id: chance.integer().toString(),
  created_at: chance.date().toString(),
  updated_at: chance.date().toString(),
  name: chance.name(),
  status: chance.pickone(Object.values([StateEnum.DEPLOYED, StateEnum.RUNNING, StateEnum.STOP_ERROR])),
}

const database = {
  id: chance.integer().toString(),
  created_at: chance.date().toString(),
  updated_at: chance.date().toString(),
  name: chance.name(),
  status: chance.pickone(Object.values([DeploymentHistoryStatusEnum.SUCCESS, DeploymentHistoryStatusEnum.FAILED])),
}

const deployment = {
  id: chance.integer().toString(),
  created_at: chance.date().toString(),
  updated_at: chance.date().toString(),
  status: StateEnum.DEPLOYED,
  applications: [service, service, service],
  databases: [database, database, database],
}

const deploymentRules = {
  id: chance.guid(),
  created_at: chance.date().toString(),
  updated_at: chance.date().toString(),
  auto_deploy: true,
  auto_stop: true,
  auto_delete: true,
  auto_preview: true,
  timezone: 'UTC',
  start_time: chance.date().toString(),
  stop_time: chance.date().toString(),
  weekdays: [chance.pickone(Object.values(WeekdayEnum))],
}

export const environmentFactoryMock = (howMany: number, noStatus = false, noDeployments = false): EnvironmentEntity[] =>
  Array.from({ length: howMany }).map((_, index) => ({
    id: `${(index + 1) * Math.random()}`,
    created_at: chance.date().toString(),
    updated_at: chance.date().toString(),
    name: chance.name(),
    last_updated_by: chance.date().toString(),
    cloud_provider: {
      provider: chance.name(),
      cluster: chance.name(),
    },
    mode: chance.pickone(
      Object.values([EnvironmentModeEnum.DEVELOPMENT, EnvironmentModeEnum.PRODUCTION, EnvironmentModeEnum.STAGING])
    ),
    cluster_id: chance.guid(),
    status: !noStatus
      ? {
          id: `${index}`,
          state: chance.pickone(Object.values([StateEnum.DEPLOYED, StateEnum.RUNNING, StateEnum.STOP_ERROR])),
          message: chance.word({ length: 10 }),
          service_deployment_status: chance.pickone(Object.values([ServiceDeploymentStatusEnum.UP_TO_DATE])),
        }
      : undefined,
    deployments: !noDeployments ? [deployment, deployment, deployment] : undefined,
    deploymentRules: deploymentRules,
  }))
