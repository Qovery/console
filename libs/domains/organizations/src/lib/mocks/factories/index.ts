export * from './organizations-factory.mock'
