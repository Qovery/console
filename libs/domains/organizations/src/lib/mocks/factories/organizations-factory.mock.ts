import { OrganizationInterface } from '../../interfaces/organizations.interface'
import { OrganizationPlanType } from '../../enums/organizations.enum'
import { Chance } from 'chance'

const chance = new Chance()

export const organizationsFactoryMock = (howMany: number): OrganizationInterface[] =>
  Array.from({ length: howMany }).map((_, index) => ({
    id: `${index}`,
    created_at: new Date(),
    updated_at: new Date(),
    name: chance.name(),
    description: chance.word({ length: 10 }),
    plan: chance.pickone(Object.values(OrganizationPlanType)),
    website_url: chance.url(),
    repository: chance.name(),
    logo_url: chance.url(),
    icon_url: chance.url(),
    owner: chance.name(),
  }))
