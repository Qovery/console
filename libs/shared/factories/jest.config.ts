/* eslint-disable */
export default {
  displayName: 'shared-factories',
  preset: '../../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../coverage/libs/shared/factories',
}
