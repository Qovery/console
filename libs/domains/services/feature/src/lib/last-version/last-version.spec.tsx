import { containerFactoryMock } from '@qovery/shared/factories'
import { renderWithProviders } from '@qovery/shared/util-tests'
import LastVersion from './last-version'

const mockContainer = containerFactoryMock(1)[0]

describe('LastVersion', () => {
  it('should match snapshot', () => {
    const { baseElement } = renderWithProviders(
      <LastVersion organizationId="1" projectId="2" service={mockContainer} version="1.0" />
    )
    expect(baseElement).toMatchSnapshot()
  })
})
