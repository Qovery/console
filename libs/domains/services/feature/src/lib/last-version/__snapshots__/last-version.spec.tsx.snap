// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LastVersion should match snapshot 1`] = `
<body>
  <div>
    <span
      class="flex"
    >
      <span
        class="inline-flex items-center shrink-0 font-medium text-xs h-6 px-2 border rounded bg-neutral-100 text-neutral-400 min-w-7 max-w-[81px] rounded-r-none border-r-0 border-neutral-250"
      >
        <span
          class="flex h-full w-full items-center justify-center truncate"
        >
          1.0
        </span>
      </span>
      <button
        class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-brand-500 text-xs h-6 rounded border border-neutral-250 dark:border-neutral-400 text-neutral-400 dark:text-white hover:[&:not(:active)]:border-neutral-300 dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none bg-neutral-100 dark:bg-neutral-500 hover:[&:not(:active):not(:disabled)]:bg-white dark:hover:[&:not(:active):not(:disabled)]:bg-neutral-400 w-7 justify-center gap-1 rounded-l-none px-1.5"
        data-state="closed"
        type="button"
      >
        <span
          class="flex h-full items-center justify-center"
        >
          <i
            aria-hidden="true"
            class="fa-regular fa-clock-rotate-left "
          />
        </span>
      </button>
    </span>
  </div>
</body>
`;
